doc/manual/packages.rst

```diff
@@ -621,11 +629,22 @@ not domain experts*.
 
   * ``MATLAB.jl`` provides an interface to call the MATLAB engine from within Julia.
 
+6. Creative names are cool, but confusion isn't so much.
+
+  * Many modern programming languages such as Ruby and Python have popular packages
+    that have entertaining names. Python's ``pickle`` package isn't about pickles, it's
+    for serializing (or pickling) object structures. Ruby's ``Sinatra`` package is named
+    after the popular American singer and is a popular web framework that competes with Ruby's
+    ``Rails`` web framework (another creative name).
+
+  * If being a creative package namer isn't your style, it's okay to not force it.
```

Comment:
This sounds like you should find a funny name, and only use a boring one if you're not smart enough. Also the first paragraph isn't clear about whether you consider this a good idea or not. I'd prefer people to use descriptive names where possible instead of what you explain about `pickle` and Sinatra`, and AFAICT this was the logic of the existing text. how about saying "use a descriptive package name instead of a creative one where applicable to help users finding out about it".